distutils/core.py

```diff
@@ -205,14 +216,17 @@ def run_setup (script_name, script_args=None, stop_after="run"):
     _setup_stop_after = stop_after
 
     save_argv = sys.argv.copy()
-    g = {'__file__': script_name}
+    g = {'__file__': script_name, '__name__': '__main__'}
     try:
         try:
             sys.argv[0] = script_name
             if script_args is not None:
                 sys.argv[1:] = script_args
-            with open(script_name, 'rb') as f:
-                exec(f.read(), g)
+            _open = getattr(tokenize, 'open', open)
```

Comment:
If Python < 3.2 is not supported, this `getattr` could also be dropped.
I was not sure about which version `distutils` is targetting.

Reply:
It should be the same as setuptools, so this could be dropped.